### ISSUE TITLE
[DOC] Produce notebook showing how to normalize and standardize columns using pyjanitor

### ISSUE THREAD

**Issue by jcvall** [ericmjl/pyjanitor#135]
I like to add a normalize and standardize function to pyjanitor. I guess we can do this with transform function, but I like the readability of the code over a lambda function to make it work. Like normalize_columns() or standardize_columns(), since we already have expand_columns() (aka. dummy codes) and max_min_scale(). This will put preprocessing into the hands of more beginners, even though I have nothing against sklearn. I will say, doing a .fit() call on preprocessing bothers me, and I dont know why? What are your thoughts?

**Comment by ericmjl** [comment]
@jcvall I debated a bit with myself when thinking about whether to put min_max_scale and expand_columns inside. I think it's cool to put in `normalize`/`standardize`, but I think that it (and now that I'm thinking about it, `min_max_scale` as well) should come with a warning in the docs, that state that any transformations that involve a summary statistic of a dataset (e.g. mean, median, max, min, etc.) will result in testing set "contamination", or "leakage of information from training set into testing set". For this reason, sklearn's preprocessing is instead more principled.

There is one setting where I think having those functions could be useful though - in Bayesian modelling, when we are already sure of the model structure, then fitting the model necessarily means using the entire dataset, hence no train/test splitting happens. Here, having these functions could be useful as well.

Even in light of these statistical considerations, I would encourage you to submit the PR! We can probably do it with the right caveats written into the docstring, so that other end-users are aware of how to not abuse the convenience functionality. What do you think?

**Comment by jcvall** [comment]
I actually didn't think about leakage in machine learning.  I guess what you are saying is that when you perform the preprocessing  you can split your data; however, the deviations from the mean for normalized data will be in both the training and test set.  So the mean is inferred in both data sets.  If I am thinking correctly, then that is a problem...Let me know if I am off base here.  I guess in Bayesian and more classical statistics (linear regression) it is rather helpful function for the coefficients, so I see your point.  Okay.  I will go ahead  knowing that this should be used with caution.

**Comment by ericmjl** [comment]
Yep, you're on the right track there!

**Comment by jcvall** [comment]
Sorry, you don't have to answer this right away, but would the sklearn-pandas package be helpful here?  

Example from GitHub page:
`Outputting a dataframe

By default the output of the dataframe mapper is a numpy array. This is so because most sklearn estimators expect a numpy array as input. If however we want the output of the mapper to be a dataframe, we can do so using the parameter df_out when creating the mapper:

mapper_df = DataFrameMapper([
('pet', sklearn.preprocessing.LabelBinarizer()),
['children'], sklearn.preprocessing.StandardScaler())
], df_out=True)
np.round(mapper_df.fit_transform(data.copy()), 2)
   pet_cat  pet_dog  pet_fish  children
0        1        0         0      0.21
1        0        1         0      1.88
2        0        1         0     -0.63
3        0        0         1     -0.63
4        1        0         0     -1.46
5        0        1         0     -0.63
6        1        0         0      1.04
7        0        0         1      0.21`

**Comment by ericmjl** [comment]
I gave it some thought last night and this morning.

It could be good! We already use scikit-learn underneath the hood, I think. 

One thing I did was to profile how fast using sklearn was compared to a custom implementation of one-hot encoding, and I found my custom implementation was faster. That said, for convenience, I ended up just using the get_dummies function provided by pandas. Maybe measuring function speed could be helpful here in deciding which implementation to use?

**Comment by jcvall** [comment]
I am sort of now leaning toward doing the documentation of both methods.  On with the normalize_columns and standardize_columns with Bayesian or classical statistics.  And maybe one with sklean using a pipe function (I have actually not done this in practice, I will try it).  That way users can have options that when they read the documentation.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jcvall** [comment]
I like to close this one as I have really learn to use your transform_columns function for this.  I also love to use the transform_column to edit text stings with a lambda s: s.str.replace().  It works great.  What are your thoughts? Maybe we show examples of this in the documentation?  I can maybe help provide you a couple examples if you don’t have some ready?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ericmjl** [comment]
@jcvall examples are amazing! I would love to have them added to pyjanitor's documentation. One of the "curses" of being a package developer is that we get too familiar with the package to write good examples. Would you like to get a few started?

**Comment by jcvall-olympus** [comment]
Yes!

**Comment by jcvall** [comment]
[pyjanitor_normalize.pdf](https://github.com/ericmjl/pyjanitor/files/3129933/pyjanitor_normalize.pdf)

not sure what form to send it in.  Here is pdf....

**Comment by ericmjl** [comment]
@jcvall I like what you did! I think you can do the following to make it a contribution to pyjanitor's documentation:

1. Perhaps annotate which functions you used are `pyjanitor` functions?
2. Add some prose that puts in plain English what you're trying to do?
3. Submit a PR with the notebook! (The best place for this is probably under `docs/` for now.) Once it's PR-d in, I'll find a way to include it as part of the hosted documentation.

**Comment by catherinedevlin** [comment]
I think I can produce this as requested from @jcvall 's PDF if needed!

**Comment by ericmjl** [comment]
Thank you, @catherinedevlin, this would be a welcome contribution!
